### ISSUE TITLE
Is there race condition in the ego when call enclave function?

### ISSUE THREAD

**Issue by therainisme** [edgelesssys/ego#229]
### Issue description

I write a test function when I want to test the enclave local attestation performance. My code is modified based on LocalAttestation. But when program executes for a period of time, the server or client will produce segmentation fault. However, after I comment the `verifyReport` function of server or client, this error doesn't occur.

### To reproduce

1. Modify `httpGet` function of client, add timeout. Avoid the (Client.Timeout exceeded while awaiting headers) error.

```
client := http.Client{Transport: &http.Transport{TLSClientConfig: tlsConfig}, Timeout: 100 * time.Second}
```

2. Modify client's origin main function to test function

```go
func test(threads, count int) {
	for c := 0; c < count; c++ {
		wg := sync.WaitGroup{}

		for t := 0; t < threads; t++ {
			wg.Add(1)
			go func() {
				do() // origin main function
				wg.Done()
			}()
		}
		wg.Wait()
	}
}

func main() {
	test(200, 20)
}
```

### Expected behavior

Server or client may produce segmentation fault.

### Additional info

Here are some fault messages about openenclave-src.

- server

```
~/workspace/go-sgx-experiment/TestLocalAttestation/server$ ego run server 
EGo v1.4.1 (8b99356398dd3bcb5f74e5194d20ce421f607404)
[erthost] loading enclave ...
[erthost] entering enclave ...
[ego] starting application ...
listening ...
ERROR: Segmentation fault [openenclave-src/enclave/core/sgx/exception.c:oe_real_exception_dispatcher:469]
ERROR: Backtrace:
ERROR: net/http.(*Server).trackConn(): 0x7fdfc10f2021
ERROR: net/http.(*conn).setState(): 0x7fdfc10ec8f5
ERROR: net/http.(*conn).serve.func1(): 0x7fdfc10ee0a8
ERROR: runtime.deferreturn(): 0x7fdfc0ef02d3
ERROR: net/http.(*conn).serve(): 0x7fdfc10ed36e
ERROR: net/http.(*Server).Serve.func3(): 0x7fdfc10f174e
ERROR: runtime.goexit.abi0(): 0x7fdfc0f228c1
ERROR: Backtrace:
ERROR: Segmentation fault [openenclave-src/enclave/core/sgx/exception.c:oe_real_exception_dispatcher:469]
ERROR: Backtrace:
ERROR: oe_abort_with_td(): 0x7fdfc087eab3
ERROR: oe_abort(): 0x7fdfc087d898
ERROR: oe_real_exception_dispatcher(): 0x7fdfc0880fce
ERROR: net/http.(*Server).trackConn(): 0x7fdfc10f2021
ERROR: net/http.(*conn).setState(): 0x7fdfc10ec8f5
ERROR: net/http.(*conn).serve.func1(): 0x7fdfc10ee0a8
ERROR: runtime.deferreturn(): 0x7fdfc0ef02d3
ERROR: net/http.(*conn).serve(): 0x7fdfc10ed36e
ERROR: net/http.(*Server).Serve.func3(): 0x7fdfc10f174e
ERROR: runtime.goexit.abi0(): 0x7fdfc0f228c1
ERROR: :OE_ENCLAVE_ABORTING [openenclave-src/host/calls.c:_call_enclave_function_impl:56]
ERROR: signal: aborted
```

- client

```
~/workspace/go-sgx-experiment/TestLocalAttestation/client$ ego run client
EGo v1.4.1 (8b99356398dd3bcb5f74e5194d20ce421f607404)
[erthost] loading enclave ...
[erthost] entering enclave ...
[ego] starting application ...
ERROR: Segmentation fault [openenclave-src/enclave/core/sgx/exception.c:oe_real_exception_dispatcher:469]
ERROR: Backtrace:
ERROR: net/http.(*Transport).roundTrip(): 0x7f1cc10d6636
ERROR: net/http.(*Transport).RoundTrip(): 0x7f1cc10cdbd9
ERROR: net/http.send(): 0x7f1cc10abc17
ERROR: net/http.(*Client).send(): 0x7f1cc10ab49b
ERROR: net/http.(*Client).do(): 0x7f1cc10ad75b
ERROR: net/http.(*Client).Get(): 0x7f1cc10acd2a
ERROR: main.httpGet(): 0x7f1cc10ef0b5
ERROR: main.do(): 0x7f1cc10ee8ab
ERROR: main.test.func1(): 0x7f1cc10ef4e5
ERROR: runtime.goexit.abi0(): 0x7f1cc0f21441
ERROR: Backtrace:
ERROR: oe_abort_with_td(): 0x7f1cc087eab3
ERROR: oe_abort(): 0x7f1cc087d898
ERROR: oe_real_exception_dispatcher(): 0x7f1cc0880fce
ERROR: net/http.(*Transport).roundTrip(): 0x7f1cc10d6636
ERROR: net/http.(*Transport).RoundTrip(): 0x7f1cc10cdbd9
ERROR: net/http.send(): 0x7f1cc10abc17
ERROR: net/http.(*Client).send(): 0x7f1cc10ab49b
ERROR: net/http.(*Client).do(): 0x7f1cc10ad75b
ERROR: net/http.(*Client).Get(): 0x7f1cc10acd2a
ERROR: main.httpGet(): 0x7f1cc10ef0b5
ERROR: main.do(): 0x7f1cc10ee8ab
ERROR: main.test.func1(): 0x7f1cc10ef4e5
ERROR: runtime.goexit.abi0(): 0x7f1cc0f21441
ERROR: :OE_ENCLAVE_ABORTING [openenclave-src/host/calls.c:_call_enclave_function_impl:56]
ERROR: signal: aborted
```

**Comment by thomasten** [comment]
Fixed in v1.4.3. This is an update to the EGo API package only. Please update your go.mod.

**Comment by therainisme** [comment]
Thank you so much for addressing the issue promptly!
